ui/app/routes/vault/cluster/secrets/backend/configuration/index.js

```diff
@@ -5,28 +5,132 @@
+        ...configModels,
+      });
+    }
+    return hash({ secretEngineModel });
+  }
+
+  standardizeConfigModels(configModels) {
+    // standardize the configModels to an array so that the component can handle it correctly
+    Array.isArray(configModels) ? configModels : (configModels = [configModels]);
+    // make sure no items in the array are null or undefined
+    configModels.forEach((configModel) => {
+      if (!configModel) {
+        configModels.splice(configModels.indexOf(configModel), 1);
       }
+    });
+
+    return configModels;
+  }
+
+  async fetchConfig(type, id) {
+    switch (type) {
+      case 'aws':
+        return await this.fetchAwsConfigs(id);
+      case 'ssh':
+        return await this.fetchSshCaConfig(id);
     }
-    return backend;
+  }
+
+  async fetchAwsConfigs(id) {
+    // AWS has two configuration endpoints root and lease, return an array of these responses.
+    const configArray = [];
+    const configRoot = await this.fetchAwsRoot(id);
+    const configLease = await this.fetchAwsLease(id);
+    configArray.push(configRoot, configLease);
+    return configArray;
+  }
+
+  async fetchAwsLease(id) {
+    try {
+      return await this.store.queryRecord('aws/lease-config', { backend: id });
+    } catch (e) {
+      if (e.httpStatus === 404) {
+        // a 404 error is thrown when the lease config hasn't been set yet.
+        return;
+      }
+      throw e;
+    }
+  }
+
+  async fetchAwsRoot(id) {
+    try {
+      return await this.store.queryRecord('aws/root-config', { backend: id });
+    } catch (e) {
+      if (e.httpStatus === 404) {
+        // a 404 error is thrown when the root config hasn't been set yet.
+        // return and let the component handle the empty config.
+        return;
+      }
+      throw e;
+    }
+  }
+
+  async fetchSshCaConfig(id) {
+    try {
+      return await this.store.queryRecord('ssh/ca-config', { backend: id });
+    } catch (e) {
+      if (e.httpStatus === 400 && e.errors[0] === `keys haven't been configured yet`) {
+        // When first mounting a SSH engine it throws a 400 error with this specific message.
+        // We want to catch this situation and return nothing so that the component can handle it correctly.
+        return;
+      }
+      throw e;
+    }
+  }
+
+  setupController(controller, resolvedModel) {
+    super.setupController(controller, resolvedModel);
+    controller.typeDisplay = allEngines().find(
+      (engine) => engine.type === resolvedModel.secretEngineModel.type
+    )?.displayName;
+    controller.isConfigurable = CONFIGURABLE_SECRET_ENGINES.includes(resolvedModel.secretEngineModel.type);
+    controller.modelId = resolvedModel.secretEngineModel.id;
+    // from the resolvedModel remove the secretEngineModel as it's not needed in the configuration details component
+    const configModels = { ...resolvedModel };
+    delete configModels.secretEngineModel;
+    controller.configModels = Object.values(configModels);
```

Comment:
`configModels` is already an array, right? What is this intended to do?

Reply:
It's an object at this point. I'll add in a clarifying comment.
![image](https://github.com/user-attachments/assets/dc856aaf-b278-49ac-a2e0-c2dcbb27bd5f)

Reply:
Good catch. A little mix and match of an old approach that is no longer needed.